feat: Detect and report permissions and properties requiring a privileged signature (#4307)
mozilla/addons-linter: docs/rules.md

| `MANIFEST_FIELD_PRIVILEGEDONLY`                         | warning  | A manifest field ignored on non-privileged add-ons                                              |
| `MANIFEST_FIELD_UNSUPPORTED`                            | warning  | A manifest field is not supported (or not supported for the extension manifest_version)         |
| `MANIFEST_PERMISSION_UNSUPPORTED`                       | warning  | A manifest permission is not supported for the extension manifest_version                       |
| `MANIFEST_PERMISSIONS_PRIVILEGED`                       | error    | A manifest permission only allowed in add-ons signed as privileged                              |
| `MOZILLA_ADDONS_PERMISSION_REQUIRED`                    | error    | "mozillaAddons" permission is a mandatory requirement for signing add-ons as privileged         |
| `PRIVILEGED_FEATURES_REQUIRED`                          | error    | Privileged features are mandatory for privileged add-ons, but none has been detected            |

> | `MANIFEST_FIELD_PRIVILEGED`                             | error    | A manifest field only allowed in add-ons signed as privileged                                   |

Reviewer: ```suggestion
| `MANIFEST_FIELD_PRIVILEGED`                             | error    | A manifest field is only allowed in privileged extensions                                       |
```
